pkg/run/run.go

```diff
@@ -0,0 +1,236 @@
+package run
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/fluxcd/flux2/pkg/manifestgen/install"
+	runclient "github.com/fluxcd/pkg/runtime/client"
+	"github.com/fluxcd/pkg/ssa"
+	coretypes "github.com/weaveworks/weave-gitops/core/server/types"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetFluxVersion(log logger.Logger, ctx context.Context, kubeClient *kube.KubeHTTP) (string, error) {
+	log.Actionf("Getting Flux version...")
+
+	listResult := unstructured.UnstructuredList{}
+
+	listResult.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Namespace",
+	})
+
+	listOptions := ctrlclient.MatchingLabels{
+		coretypes.PartOfLabel: "flux",
+	}
+
+	u := unstructured.Unstructured{}
+
+	if err := kubeClient.List(ctx, &listResult, listOptions); err != nil {
+		log.Failuref("error getting list of objects")
+		return "", err
+	} else {
+		for _, item := range listResult.Items {
+			if item.GetLabels()[flux.VersionLabelKey] != "" {
+				u = item
+				break
+			}
+		}
+	}
+
+	labels := u.GetLabels()
+	if labels == nil {
+		return "", fmt.Errorf("error getting labels")
+	}
+
+	fluxVersion := labels[flux.VersionLabelKey]
+	if fluxVersion == "" {
+		return "", fmt.Errorf("no flux version found")
+	}
+
+	return fluxVersion, nil
+}
+
+func InstallFlux(log logger.Logger, ctx context.Context, kubeClient *kube.KubeHTTP, kubeConfigOptions genericclioptions.RESTClientGetter) error {
+	log.Actionf("Installing Flux...")
+
+	opts := install.Options{
+		BaseURL:      install.MakeDefaultOptions().BaseURL,
+		Version:      "v0.31.2",
+		Namespace:    "flux-system",
+		Components:   []string{"source-controller", "kustomize-controller", "helm-controller", "notification-controller"},
+		ManifestFile: "flux-system.yaml",
+		Timeout:      5 * time.Second,
+	}
+
+	manifest, err := install.Generate(opts, "")
```

Comment:
This variable would be `manifests` to be consistent with your error logs

Reply:
Fixed, good catch!